File: clients/javascript/src/resources/transactions.ts
import type { HieroProxyClient } from "../client";
import type { TransactionInfoResponse, TransactionFilterOptions } from "./transactions.types";
import { requireId } from "./validation";

/**
 * Transactions resource — query transactions by ID or account.
 */
export class TransactionsResource {
  constructor(private readonly client: HieroProxyClient) {}

  /**
   * Get a transaction by its ID.
   */
  async getById(transactionId: string): Promise<TransactionInfoResponse> {
    const id = requireId(transactionId, "transactionId");
    return this.client.get(`/api/v1/transactions/${id}`);
  }

  /**
   * Get transactions for an account with optional filters.
   */
  async getByAccount(accountId: string, options?: TransactionFilterOptions): Promise<TransactionInfoResponse[]> {
    const id = requireId(accountId, "accountId");
    const params = new URLSearchParams();
    if (options?.type) params.set("type", options.type);
    if (options?.result) params.set("result", options.result);
    if (options?.modification) params.set("modification", options.modification);
    const query = params.toString();
    const path = `/api/v1/transactions/account/${id}${query ? `?${query}` : ""}`;
    return this.client.get(path);
  }
}

File: clients/python/src/hiero_proxy/resources/transactions.py
"""Transactions resource."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from hiero_proxy.resources.validation import require_id

if TYPE_CHECKING:
    from hiero_proxy.client import AsyncHieroProxyClient, HieroProxyClient


class TransactionsResource:
    """Transaction queries — get by ID or account."""

    def __init__(self, client: HieroProxyClient | AsyncHieroProxyClient) -> None:
        self._client = client

    def get_by_id(self, transaction_id: str) -> Any:
        tid = require_id(transaction_id, "transaction_id")
        return self._client.get(f"/api/v1/transactions/{tid}")

    def get_by_account(
        self,
        account_id: str,
        *,
        type: str | None = None,
        result: str | None = None,
        modification: str | None = None,
    ) -> Any:
        aid = require_id(account_id, "account_id")
        params: list[str] = []
        if type:
            params.append(f"type={type}")
        if result:
            params.append(f"result={result}")
        if modification:
            params.append(f"modification={modification}")
        query = f"?{'&'.join(params)}" if params else ""
        return self._client.get(f"/api/v1/transactions/account/{aid}{query}")
